### ISSUE TITLE
Redesign Join page

### ISSUE THREAD

**Issue by mikerod-sd** [OperationCode/front-end#1124]
# Feature request

Add wording on https://operationcode.org/join that explains why we collect demographic information

Also add wording on next page after submit (or whatever page is returned after final submit) that outlines how to join slack (i.e. "Thanks for submitting your information, please be on the lookout for an email which will have a link to join our slack")

## Proposed solution

See above.

## Potential alternative solutions

N/A

## Additional context

Per discussion in the mod channel on slack.

**Comment by Too-Far** [comment]
I can tackle this if you guys want! I should be able to do it Tuesday night if not sooner.

**Comment by datphotogurl** [comment]
Adding continued functionality requests to the https://operationcode.org/join page:

1. Add checkbox and new window expansion for "I have read the Operation Code's _**Code of Conduct**_ and agree to abide by it during my time as a member."
2. Add checkbox and new window expansion of rules "I agree to adhere to _**Slack's Community Guidelines**_"

Screenshot of La Techqueria's Join page is attached for an example.
<img width="670" alt="Screen Shot 2020-07-30 at 3 40 03 PM" src="https://user-images.githubusercontent.com/51416206/88967000-31690d00-d27b-11ea-864b-4ab6805a3822.png">

**Comment by kylemh** [comment]
No back-end needed. We'll just do front-end validation regarding these checks since they're not legal in nature.

**Comment by kylemh** [comment]
@datphotogurl can we get links to the two policies? I'd consider this ready to be worked on if we have those links (assuming we'll copy... their copy)

**Comment by datphotogurl** [comment]
> @datphotogurl can we get links to the two policies? I'd consider this ready to be worked on if we have those links (assuming we'll copy... their copy)

Hi @kylemh, I was walking through this issue with Lianis as an exercise, but yes, I can add the two sets of content. Currently, the Code of Conduct is being updated by the Board and Mods. 

The link for the Code of Conduct: https://github.com/OperationCode/operationcode_docs/blob/master/community/code_of_conduct.md 

We do not currently have a repo for a Slack Community Guideline but I can write up that content and provide that to you to either embed in Github or the website. Which do you suggest?

**Comment by kylemh** [comment]
I would just make a new markdown file in that same repo!

**Comment by kylemh** [comment]
@datphotogurl can we get agreed upon content for this?

**Comment by datphotogurl** [comment]
Yes, I need to work on it on the weekend as my weekday schedule is ramped
up now.

*Cynthia S. Kao,*
Founder and Lead Producer,
*Rebirth Media LLC*
+1(202)716-5860
https://ckaojohnson.wix.com/rebirthmedia


* <https://www.facebook.com/RebirthMediaLLC>
<https://instagram.com/datphotogurl/> <https://twitter.com/datphotogurl>*

*Notice of Confidentiality:The information transmitted is intended only for
the person or entity to which it is addressed and may contain confidential
and/or privileged material. Any review re-transmission dissemination or
other use of or taking of any action in reliance upon this information by
persons or entities other than the intended recipient is prohibited. If you
received this in error please contact the sender immediately by return
electronic transmission and then immediately delete this transmission
including all attachments without copying distributing or disclosing same.*

**Comment by ksmacleod99** [comment]
Slack community guidelines: https://github.com/OperationCode/START_HERE/blob/master/community_guidelines.md
(thanks conrad!) 

if I can get mocks, I will claim this. 

if I can't get mocks, I will claim this and it will look fugly :)

**Comment by kylemh** [comment]
Go for it and make it extra ugly

**Comment by ksmacleod99** [comment]
oh hey did we want a reCaptcha like La Techqueria?

**Comment by mikerod-sd** [comment]
Do we have a problem with bots? If not I say don't implement one.

**Comment by kylemh** [comment]
We had a PR open for a captcha awhile ago, but yeah... i think we can assume nobody cares enough to bot us.

**Comment by ksmacleod99** [comment]
Just going to wait on the formik upgrade PR to get merged :)

**Comment by ksmacleod99** [comment]
@kylemh  would you rather see a new checkbox component, or just keep the checkbox local to the registration form?

**Comment by kylemh** [comment]
I guess we don't have a checkbox component. I think we can keep it simple by adding CSS to globalStyles to add styles to `input[type="checkbox"]`

**Comment by LindaCBeall** [comment]
I guess I just posted this mock on slack but not here, @datphotogurl did you have a chance to review this yet?
https://xd.adobe.com/view/4b138861-8af9-474d-87e4-ba8142b7f762-1e40/
<img width="716" alt="Screen Shot 2020-10-07 at 2 13 05 PM" src="https://user-images.githubusercontent.com/37910748/95388510-3d8acf80-08a7-11eb-90ec-3da8b0e8ec4f.png">

**Comment by kylemh** [comment]
@LindaCBeall I think the above can be added to a ticket to redesign the whole join page entirely. This issue is just to get people to accept the rules before they register at a bare minimum.

**Comment by datphotogurl** [comment]
@kylemh @LindaCBeall I think further discussion needs to be warranted for an entire new design of that page, and probably not something we can have done for Hacktoberfest. More ideas on this but prefer to discuss possibilities for the new design at Product Meeting. We previously discussed the need to have two checkboxes for language that states _I have read and acknowledge adherence to the Operation Code Code of Conduct_ and _I have read and acknowledge adherence to the Operation Code Slack Community Guidelines_. Those smaller items are more time sensitive and needed to cover us for legal purposes.

**Comment by kylemh** [comment]
Definitely; however, I'm also sure there's still work we can do now regarding the original post.

**Comment by datphotogurl** [comment]
Agree with you Kyle. I just realized there isn't actual verbiage when Mike opened the issue. Here's the language on that:
_The demographic information you provide us helps us understand our community needs, ensure diversity and provide specific resources to reach our mission. Thank you in advance for providing honest answers._

**Comment by kylemh** [comment]
@datphotogurl perfect. Any idea _where_ you want that rendered on `/join`?

**Comment by datphotogurl** [comment]
Right before the "submit button" in place of this wording currently on the page: The information we collect is to help us personalize your experience on our Slack community. We do not sell your information to anyone.

**Comment by LindaCBeall** [comment]
@ksmacleod99 are you still interested in working on this?

**Comment by ksmacleod99** [comment]
It's been so long I hardly remember this -- you or Kyle can un-assign me

**Comment by calincionca35** [comment]
What would be the **current rephrasing of the _specifics_ of the original issue** that would take into consideration the changes discussed in the above conversation?

**Comment by Lianis-Operations** [comment]
Created separate ticket to add verbiage to "Join" page stating why we collect demographic information #1426

**Comment by kylemh** [comment]
@datphotogurl lemme know how you'd like the OP of this issue to look to better reflect the title.

**Comment by friendlymatthew** [comment]
Hi is this issue available to work on?

**Comment by kylemh** [comment]
Hey @friendlymatthew yes. Just please be sure to read the entire conversation. This is a really small ticket and not at all a full redesign of the page!
